codalab/lib/beam/MultiReaderFileStream.py

```diff
@@ -1,21 +1,22 @@
-from io import BytesIO
+from io import BytesIO, SEEK_SET, SEEK_END
 from threading import Lock
-
-from codalab.worker.un_gzip_stream import BytesBuffer
-
-
+import time
 class MultiReaderFileStream(BytesIO):
     """
-    FileStream that support multiple readers
+    FileStream that support multiple readers and seeks backwards
     """
     NUM_READERS = 2
+    LOOKBACK_LENGTH = 33554432
+    MAX_THRESHOLD = LOOKBACK_LENGTH * 2
 
     def __init__(self, fileobj):
-        self._bufs = [BytesBuffer() for _ in range(0, self.NUM_READERS)]
-        self._pos = [0 for _ in range(0, self.NUM_READERS)]
+        self._buffer = bytes()
+        self._buffer_pos = 0 # start position of buffer in the fileobj (min reader position - LOOKBACK LENGTH)
+        self._size = 0 # size of bytes (for convenience)
```

Comment:
call it `_buffer_size` to be more unambiguous

Reply:
Actually, can we get rid of this and just use `len(_buffer)`...avoid having extra state that you have to keep in sync